alivee/place_search_engine/utils/place_search.py

```diff
@@ -0,0 +1,233 @@
+import googlemaps
+import datetime
+import json
+import time
+from place_search_engine.constants.google_place_api import PlaceSearchResponseStatus
+from place_search_engine.constants.google_place_api import PLACE_SERVICE_SEARCH_PLACE_TYPES
+from place_search_engine.constants.google_place_api import MAX_PLACE_SERACH_RADIUS_IN_METERS
+
+MAX_NEXT_PAGE_TOKEN_RETRY_COUNT = 3
+
+
+class PlaceSearch(object):
+	'''
+	Wrapped google place search service for our usage.
+	Documents: https://developers.google.com/places/web-service/intro
+
+	Usage example: Search all banks in 1000 meters radius area of location (37.7845782, -122.3910999)
+
+	placeSearchInstance = PlaceSearch(
+		location=(37.7845782, -122.3910999), 
+		key=key,
+	)
+
+	placeSearchInstance.nearbySearch(
+		place_type=PlaceServicePlaceType.BANK, 
+		radius=1000,
+	)
+	OR:
+	placeSearchInstance.radarSearch(
+		place_type=PlaceServicePlaceType.BANK, 
+		radius=1000,
+	)
+	'''
+	def __init__(self, location, key):
+		'''
+		Init method of place search class.
+
+		param location: A tuple (lat, lng) represents the location we want to set as the center of the search area.
+		param key: The Google API key, generated in google dev console, ask project dev for help.
+		'''
+		self.location = location
+		# NOTE: Need to handle exception in caller's method such as `ValueError: Invalid API key provided.`
+		self.client = googlemaps.Client(key)
+
+	def _getPlaceFilterKwargs(self, place_type, radius, keyword, name, rank_by, page_token):
```

Comment:
are these all required kwargs? If not, shall we make them as optional kwargs: `place_type=None`, `radius=None` ... ?